### PR TITLE
Remove metaCanonicalPath from V4 and test directly against escapePath…

### PR DESCRIPTION
**Pull request by mwu**
… and canonicalPath

### CODE REVIEW

File: lib/amazonka-core/test/Test/Amazonka/Sign/V4/Base.hs [hunk 1]
```diff
@@ -104,8 +104,7 @@ mkV4Paths genAbbrev genPath = do
           & requestService .~ svc
           & requestBody .~ Hashed aBody
 
-  (meta, _) <- base (Tag (sha256Base16 aBody)) req <$> arbitrary <*> arbitrary <*> arbitrary
-  pure (aPath, metaPath meta, metaCanonicalPath meta)
+  pure (aPath, escapedPath req, canonicalPath req)
```
Comment (mwu) [inline lib/amazonka-core/test/Test/Amazonka/Sign/V4/Base.hs:107]: The tradeoff here is the test ends up testing at a lower level, testing escapedPath and canonicalPath separately. Whether those two are actually used to compute the http request path and the sigv4 canonical path isn't covered by the tests.

File: lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs [hunk 2]
```diff
@@ -30,7 +30,6 @@ data V4 = V4
     metaPath :: Path,
     metaEndpoint :: Endpoint,
     metaCredential :: Credential,
-    metaCanonicalPath :: CanonicalPath,
```
Comment (mwu) [inline lib/amazonka-core/src/Amazonka/Sign/V4/Base.hs:33]: I'm not too fussed about adding this to the V4 record but I could go either way. Looking at the uses, it becomes completely opaque outside the Base module due to https://github.com/mwu/amazonka/blob/no-meta-canonical-path/lib/amazonka-core/src/Amazonka/Types.hs#L422 Once returned from `base`, the only thing possible/ever done with V4 is dumping it out to logs.

File: lib/amazonka-core/test/Test/Amazonka/Sign/V4/Base.hs (hunk 1 above)
Comment (endgame) [reply lib/amazonka-core/test/Test/Amazonka/Sign/V4/Base.hs:107]: I'm comfortable with that. The facet you're talking about should properly be tested by computing full signatures somewhere, but I'm happy to merge without requiring that sort of test.

**Review by endgame — APPROVED** [review]

File: lib/amazonka-core/test/Test/Amazonka/Sign/V4/Base.hs (hunk 1 above)
Comment (mwu) [reply lib/amazonka-core/test/Test/Amazonka/Sign/V4/Base.hs:107]: Yeah that makes sense.

**Pull request merged**